src/link/mod.rs

```diff
@@ -116,50 +117,54 @@ impl Import {
+                    Ok(Import { element: Name::new(directive_name)?, is_directive: true, alias: None })
                 } else {
-                    str.to_string()
-                };
-                Ok(Import { element, is_directive, alias: None })
+                    Ok(Import { element: Name::new(str.clone())?, is_directive: false, alias: None })
+                }
             },
             Value::Object(fields) => {
-                let mut name: Option<String> = None;
-                let mut alias: Option<String> = None;
+                let mut name: Option<NodeStr> = None;
+                let mut alias: Option<NodeStr> = None;
                 for (k, v) in fields {
                     match k.as_str() {
                         "name" => {
-                            name = Some(v.as_str().ok_or_else(|| {
+                            name = Some(v.as_node_str().ok_or_else(|| {
                                 LinkError::BootstrapError("invalid value for `name` field in @link(import:) argument: must be a string".to_string())
-                            })?.to_owned())
+                            })?.clone())
                         },
                         "as" => {
-                            alias = Some(v.as_str().ok_or_else(|| {
+                            alias = Some(v.as_node_str().ok_or_else(|| {
                                 LinkError::BootstrapError("invalid value for `as` field in @link(import:) argument: must be a string".to_string())
-                            })?.to_owned())
+                            })?.clone())
                         },
                         _ => Err(LinkError::BootstrapError(format!("unknown field `{k}` in @link(import:) argument")))?
                     }
                 }
                 if let Some(element) = name {
-                    let is_directive = element.starts_with('@');
-                    if is_directive {
-                        let element = element.strip_prefix('@').unwrap().to_string();
-                        if let Some(alias_str) = alias {
-                            if !alias_str.starts_with('@') {
-                                Err(LinkError::BootstrapError(format!("invalid alias '{}' for import name '{}': should start with '@' since the imported name does", alias_str, element)))?
-                            }
-                            alias = Some(alias_str.strip_prefix('@').unwrap().to_string());
+                    if let Some(directive_name) = element.strip_prefix('@') {
+                        if let Some(alias_str) = alias.as_ref() {
+                            let Some(alias_str) = alias_str.strip_prefix('@') else {
+                                return Err(LinkError::BootstrapError(format!("invalid alias '{}' for import name '{}': should start with '@' since the imported name does", alias_str, element)));
+                            };
+                            alias = Some(alias_str.into());
```

Comment:
The `None`/`Some`-ness of `.strip_prefix()` has the same information as `.starts_with()`. This is a small refactor to only use `.strip_prefix()` instead of two checks + an unwrap